Provider fix, anthropic Errorhandling if empty image file is read

rwese opened the pull request:
when the agent creates images, by screenshot or otherwise, which produces and file, which is empty, calling the antrophic provider will fail and the chat session is broken, it will not allow you to continue.

Session Export: 
[session_empty_image_.json](https://github.com/user-attachments/files/24149905/session_empty_image_.json)

The fix will check if there is any content, otherwise return a message to the agent, the test is entirely generated by the ai.

rwese commented:
_just cleaned the git-history_

rekram1-node commented:
/review

rwese commented on packages/opencode/test/provider/anthropic.test.ts:
```diff
@@ -0,0 +1,222 @@
+    description?: string
+    parameters?: Record<string, any>
+  }
+}
```
updated with the suggestion of the bot :)

rekram1-node commented:
So I looked at your session, looks like this is related to anthropic, not zen. The code changes you made are for zen api endpoints. This is probably not what you are looking to do..

rekram1-node commented:
Instead I think we should revert these changes and put the filter in this function:
https://github.com/sst/opencode/blob/e6a0a005d6c5e30054b848e931050bb82326efd8/packages/opencode/src/provider/transform.ts#L167

rwese commented:
@rekram1-node thank you, i will try fixing it there!

rwese committed:
fix: filter empty base64 images in provider transform

- Add validation in unsupportedParts to detect empty base64 image data
- Replace empty images with error text instead of sending to API
- Prevents Anthropic 'image cannot be empty' errors
- Add regression tests for empty image handling

rekram1-node commented:
/review

rekram1-node commented:
Nice!
